src/main/java/uk/gov/pay/webhooks/webhook/WebhookService.java

```diff
@@ -157,19 +156,12 @@ private boolean webhookHasSubscriptionForEvent(WebhookEntity webhook, InternalEv
                 .anyMatch(internalEventNames -> internalEventNames.contains(event.eventType()));
     }
 
-    public void deleteWebhookMessagesOlderThan() {
+    public void deleteWebhookMessages() {
         int maxAgeOfMessages = webhookMessageDeletionConfig.getMaxAgeOfMessages();
         int maxNumOfMessagesToExpire = webhookMessageDeletionConfig.getMaxNumOfMessagesToExpire();
-        
-        var webhookMessages = webhookMessageDao.getWebhookMessagesOlderThan(maxAgeOfMessages);
-        int webhookMessagesDeleted = webhookMessageDao.deleteMessages(
-                webhookMessages.stream().limit(maxNumOfMessagesToExpire));
-        LOGGER.info(format("%s webhook messages were deleted.", webhookMessagesDeleted));
-
-        var deliveryQueueEntities = webhookDeliveryQueueDao.getWebhookDeliveryQueueEntitiesOlderThan(maxAgeOfMessages);
-        int deliveryQueueEntitiesDeleted = webhookDeliveryQueueDao.deleteDeliveryQueueEntries(
-                deliveryQueueEntities.stream().limit(maxNumOfMessagesToExpire));
-        LOGGER.info(format("%s webhook delivery queue entities were deleted.", deliveryQueueEntitiesDeleted));
 
+        List<WebhookMessageEntity> webhookMessagesToDelete = webhookMessageDao.getWebhookMessagesOlderThan(maxAgeOfMessages);
```

Comment:
Any reason webhooks messages need to be retrieved and deleted? Can we not delete without querying?

Reply:
You're right, messages don't need to be retrieved.

Reply:
> Any reason webhooks messages need to be retrieved and deleted? Can we not delete without querying?

Actually this won't work because we have a `MAXIMUM_NUMBER_OF_MESSAGES_TO_EXPIRE_PER_JOB` env var and we can't use a `DELETE` in conjunction with `LIMIT` in postgresql.

Reply:
Could use something like `delete from webhook_messages where id in (select id from webhook_messages where created_date < '2022-12-31' limit 15000);`